src/reducers/vis-state-updaters.js

```diff
@@ -933,6 +933,43 @@ export const removeLayerUpdater = (state, {idx}) => {
   return updateAnimationDomain(newState);
 };
 
+/**
+ * duplicate layer
+ * @memberof visStateUpdaters
+ * @type {typeof import('./vis-state-updaters').duplicateLayerUpdater}
+ * @public
+ */
+export const duplicateLayerUpdater = (state, {idx}) => {
+  const {layers, layerData} = state;
+  const layerToDuplicate = cloneDeep(state.layers[idx]);
+  layerToDuplicate.id = `${layerToDuplicate.config.id}_copy`;
```

Comment:
this won't work if user copies the same layer twice. use `import {generateHashId} from '../utils/strings';` to get a new hash id for each duplicated layer

Reply:
Got it